plugins/extractors/postgres/postgres.go

```diff
@@ -27,25 +28,24 @@ var defaults = []string{"information_schema", "root", "postgres"}
 
 // Config holds the set of configuration options for the extractor
 type Config struct {
-	UserID   string `mapstructure:"user_id" validate:"required"`
-	Password string `mapstructure:"password" validate:"required"`
-	Host     string `mapstructure:"host" validate:"required"`
-	Database string `mapstructure:"database" default:"postgres"`
-	Exclude  string `mapstructure:"exclude"`
+	ConnectionURL string `mapstructure:"connection_url" validate:"required"`
+	Exclude       string `mapstructure:"exclude"`
 }
 
 var sampleConfig = `
-host: localhost:1433
-user_id: admin
-password: "1234"
-database: database_name
-exclude: postgres`
+connection_url: "postgres://admin:pass123@localhost:3306/testDB?sslmode=disable"
+exclude: testDB`
 
 // Extractor manages the extraction of data from the extractor
 type Extractor struct {
-	logger log.Logger
-	config Config
-	db     *sql.DB
+	logger   log.Logger
+	config   Config
+	db       *sql.DB
+	userDB   string
```

Comment:
Any thought, how we can remove `userDB` from `Extractor` struct and initialize `*sql.DB` directly.

Reply:
@scortier can you please give more context ?